### ISSUE TITLE
New districts formed in 2019

### ISSUE THREAD

**Issue by aadityadar** [guneetnarula/covid19-in#6]
Duplicate of https://github.com/covid19india/CovidCrowd/issues/7

- Local government directory (LGD) reports 11 districts in Mizoram (instead of 8; hnahthial, khawzawl, saitual are three new districts formed in June 2019)
- LGD reports 33 district in Tamil Nadu (instead of 32; kallakurichi district was formed in november 2019)
- LGD reports 20 districts in J&K (instead of 22; mirpur and muzaffarabad not reported by LGD)

- Keeping the two districts in POK, we should have 731 districts = 729 in /js/districts_729.js - 2 duplicates (assam and gujarat, see #5) + 3 mizoram + 1 tamil nadu

- If we want to be consistent with LGD there should be 729 districts (exclude 2 districts from J&K), no of districts in states should be: 
```
----------------------------------------
    State Name (In English) |      Freq.
----------------------------+-----------
andaman and nicobar islands |          3
             andhra pradesh |         13
          arunachal pradesh |         25
                      assam |         33
                      bihar |         38
                 chandigarh |          1
               chhattisgarh |         27
     dadra and nagar haveli |          1
              daman and diu |          2
                      delhi |         11
                        goa |          2
                    gujarat |         33
                    haryana |         22
           himachal pradesh |         12
          jammu and kashmir |         20
                  jharkhand |         24
                  karnataka |         30
                     kerala |         14
                     ladakh |          2
                lakshadweep |          1
             madhya pradesh |         52
                maharashtra |         36
                    manipur |         16
                  meghalaya |         11
                    mizoram |         11
                   nagaland |         11
                     odisha |         30
                 puducherry |          4
                     punjab |         22
                  rajasthan |         33
                     sikkim |          4
                 tamil nadu |         33
                  telangana |         33
                    tripura |          8
              uttar pradesh |         75
                uttarakhand |         13
                west bengal |         23
----------------------------------------
```

No of districts in js/districts_729.js is: 
```
-------------------------------
 properties/stname |      Freq.
-------------------+-----------
 andaman & nicobar |          3
    andhra pradesh |         13
 arunachal pradesh |         25
             assam |         34
             bihar |         38
        chandigarh |          1
      chhattisgarh |         27
dadra & nagar have |          1
       daman & diu |          2
             delhi |         11
               goa |          2
           gujarat |         34
           haryana |         22
  himachal pradesh |         12
   jammu & kashmir |         22
         jharkhand |         24
         karnataka |         30
            kerala |         14
            ladakh |          2
       lakshadweep |          1
    madhya pradesh |         52
       maharashtra |         36
           manipur |         16
         meghalaya |         11
           mizoram |          8
          nagaland |         11
            odisha |         30
        puducherry |          4
            punjab |         22
         rajasthan |         33
            sikkim |          4
        tamil nadu |         32
         telangana |         33
           tripura |          8
     uttar pradesh |         75
       uttarakhand |         13
       west bengal |         23
-------------------------------
```

**Comment by aadityadar** [comment]
so i took your list of 727 districts (after removing the 2 duplicates) and mapped them onto some key variables that i think should be helpful to keep like a two letter standardized state code, updated state numeric code (this resolves #7), district names that match those in local government directory/lgd, and three numeric district codes (one for 2020, 2011 and 2001 respectively; these also come from lgd). please use this file.

[district_names_clean_with_lgdcodes.xlsx](https://github.com/guneetnarula/covid19-in/files/4396963/district_names_clean_with_lgdcodes.xlsx)

**Comment by guneetnarula** [comment]
Thanks! This is great. I wasn't aware of the four new districts, let's try and look for officially notified administrative boundaries and see if we can map them manually. At this point though these haven't found their way into official directories or listings. The LG directory, NIC districts portal and the state websites do not include them consistently.

My suggestion is that we continue with 727 for now though. And consider the 4 new districts as part of their previous delimitations (to report cases)

Saitual -> Aizawl
Hnahthial -> Lunglei
Khawzawl -> Champhai
Kallakurichi -> Villuppuram

What do you think?

**Comment by aadityadar** [comment]
Yeah I guess that's one way to go about it. Btw there are few more new districts in Tamil Nadu that got formed towards the end of 2019. One got even formed in 2020 taking the total to 38. LGD reports 34 but by the end of 2019 there were 37. I checked the state media bulletins for TN and they are also reporting data for the 37 districts. I think we should just have a cutoff date, 31 Dec 2019 and use boundaries on that date. (Census of India is also using the same cutoff for the 2021 census). I can share the new districts for TN in sometime, but if you need them sooner they aren't hard to find. Wikipedia has a good list and tnrd.gov.in has notifications on the bifurcations

**Comment by guneetnarula** [comment]
Yeah they are reporting for 37. But do we have any source for the boundaries? Chengalpattu was part of Kancheepuram and carved out in July 2019 it appears, but official district boundaries are not available anywhere yet. I did see this on wikimedia https://upload.wikimedia.org/wikipedia/commons/4/46/Chengalpattu_district_location_map.svg

**Comment by aadityadar** [comment]
yep, best bet seems to use wiki for now. i've opened an issue on justin's repo, see this https://github.com/justinelliotmeyers/India_Official_Boundaries_2019/issues/4

**Comment by justinelliotmeyers** [comment]
@guneetnarula and @aadityadar I want to make sure any updates or help i can provide go to you and the people of India. I dont mind closing shop on my repo to help update here.let me know your thoughts as I want the data and assistance to go to your efforts. Justin

**Comment by aadityadar** [comment]
my two cents: it will be good to keep the repo on 2019 district boundaries separate (so that it can be used beyond just covid) + you should get credit for all that work that you are doing. it's not easy to georeference and verify boundaries!

**Comment by guneetnarula** [comment]
@justinelliotmeyers  we can put this up on datameet's git, and credit you?

**Comment by justinelliotmeyers** [comment]
@guneetnarula absolutely. I also told @thejeshgn they can do whatever they need. I am traveling now and will be home on Friday. Talk soon, thanks!
